### DISCUSSION TITLE
Suggestions for battery powered ESP32 solution.

### DISCUSSION THREAD

**Discussion by dali-dbr** [arendst/Tasmota#21573] [Q&A]
Hi. 

What would be a good choice of parts to start a battery-powered ESP32 project?

**ESP Board**
Are there any ESP32 boards that are more suitable for deep sleep than others? I was planning to use Wemos D1 mini.

**Battery**
Shelly uses 3V CR123A batteries. I suppose 3V is not good enough to power the ESP32 directly. Should I use DC/DC step-up or mini boost devices to get 5V.
https://www.adafruit.com/product/4654

**Battery status indicator in %**
How to display the battery status? Is this a simple connection of a battery + to one of the GPIOs or there is more to it?

![image](https://github.com/arendst/Tasmota/assets/6317229/09800660-b9b0-477c-a077-b86caa65cefa)
I found the pic on the web.

Thank you!

**Comment by sfromis** [comment]
Wemos D1 Mini is not an ESP32 board, but I suppose that you mean a similar ESP32-based one. 

Many dev boards are not really good for battery operation, assuming that you want to maximize battery life time with deep sleep. The board may be drawing power also while the MCU is not so much. In general, I'd recommend boards specifically designed for battery operation, including having a battery connector and charge circuit. Such boards may commonly have a analog input to check on battery voltage. Beware that there is no simple calculation from voltage to charge level, as the discharge curve is rather flat, until dropping fast at the end. Maybe some boards have a "fuel gauge" circuit for more data than just voltage.

I have no broad overview, but do know that [Unexpected Maker](https://unexpectedmaker.com/) is making some boards with focus on low power consumption. They are not the cheapest.... Others may also work ok.

**Comment by dali-dbr** [comment]
Thank you for your reply!

Yes, I meant any board that can run Tasmota...

I guess you mean boards like a [Wemos Battery Shield](https://www.wemos.cc/en/latest/d1_mini_shield/battery.html). For this board I would need 3.7V to 5V battery, but could be rechargeable via USB port.
![image](https://github.com/arendst/Tasmota/assets/6317229/c2476626-79f8-471f-b131-524d1485b3ad)

In case I don't want to have a rechargeable option. You think my idea of using 3V CR123A battery via Mini DC-DC Boost Step Up Converter is a good solution? I would assume that any conversion will take it's tool on a consumption/batery life and make PCB bigger. 

I could use 3V CR123A battery with [Step up regulator](https://www.aliexpress.com/item/1005004201092605.html) and power the board.

Could I use Wemos ADC0 (A0) input to get the voltage to Tasmota?

![image](https://github.com/arendst/Tasmota/assets/6317229/a614a04d-a1e8-443e-83da-4f64fd4f8d11)

![image](https://github.com/arendst/Tasmota/assets/6317229/9bad7969-6f89-45a9-a89b-e78ee5136cf9)

Thank you

**Reply by sfromis** [reply]
No, I did not mean adding a battery shield to a device not suitable for battery operation. The core case for that battery shield is with the D1 Mini, which is consuming power all the time. I meant a board actually suitable for battery operation, instead of a toy meant for playing around short term.

And the ESP8266 analog input is up to 1V only. The ADC0 input on the D1 Mini is up to 3.3V, and therefore not suitable for lithium batteries, unless you have a further voltage divider.

Also, barely usable cell batteries with a step-up converter is not really a "good" plan for long lifetime on battery.

Obviously, if you don't care much about battery lifetime, less need for fuzzing about details like the above....

**Reply by dali-dbr** [reply]
>The ADC0 input on the D1 Mini is up to 3.3V

Yes, the CR123A battery is 3V.

**Reply by sfromis** [reply]
Hence I added the observation that it would not be suitable for lithium batteries. The underpowered cell was only one of the options you mentioned.

**Comment by HofeBY** [comment]
I think this Board will be a good solution.
https://wiki.seeedstudio.com/xiao_esp32c6_getting_started/
Battery charger, MATTER, Thread 1.3, Zigbee 3.0, WiFi 6, BLE v5.3, USB C direct to ESP32C6

**Comment by Jason2866** [comment]
Programming the LP processor of the C6 is not supported from Arduino.
